sdk/ai/ai-inference-rest/samples-dev/streamingToolCall.ts

```diff
@@ -0,0 +1,183 @@
+      },
+    },
+    required: ["location"],
+  },
+};
+
+const getWeatherFunc = (location: string, unit: string): string => {
+  if (unit != "celsius") {
+    unit = "fahrenheit";
+  }
+  return `The temperature in ${location} is 72 degrees ${unit}`;
+}
+
+const updateToolCalls = (toolCallArray: Array<ChatCompletionsToolCallOutput>, functionArray: Array<any>) => {
+  const dummyFunction = { name: "", arguments: "", id: "" };
+  while ( functionArray.length < toolCallArray.length ) {
+    functionArray.push(dummyFunction);
+  }
+  
+  let index = 0;
+  for (const toolCall of toolCallArray) {
+    if (toolCall.function.name) {
+      functionArray[index].name = toolCall.function.name;
+    }
+    if (toolCall.id) {
+      functionArray[index].id = toolCall.id;
+    }
+    if (toolCall.function.arguments) {
+      functionArray[index].arguments += toolCall.function.arguments;
+    }
+    index++;
+  }
+}
+
+const handleToolCalls = (functionArray: Array<any>) => {
+  const messageArray = [];
+  for (const func of functionArray) {
+    const funcArgs = JSON.parse(func.arguments);
+    let content = "";
+
+    switch (func.name) {
+      
+      case "get_current_weather":
+        content =  getWeatherFunc(funcArgs.location, funcArgs.unit ?? "fahrenheit");
+        messageArray.push({
+          role: "tool",
+          content,
+          tool_call_id: func.id,
+          name: func.name
+        });
+        break;
+
+      default:
+        console.log(`unknown function ${func.name}`);
+        break;
+    }
+  }
+  return messageArray;
+}
+
+
+export async function main() {
+  const credential = new DefaultAzureCredential();
+  // auth scope for AOAI resources is currently https://cognitiveservices.azure.com/.default
+  const scopes = [ "https://cognitiveservices.azure.com/.default" ];
+  const clientOptions = { credentials: { scopes } };
+  const client = ModelClient(modelEndpoint, credential, clientOptions);
+ 
+  const messages = [{ role: "user", content: "What's the weather like in Boston?" }];
+
+  let toolCallAnswer = "";
+  let awaitingToolCallAnswer = true;
+  while (awaitingToolCallAnswer) {
+    const response = await client.path("/chat/completions").post({
+      body: {
+        messages,
+        tools: [
+          {
+            type: "function",
+            function: getCurrentWeather,
+          },
+        ],
+        stream: true
+      }
+    }).asNodeStream();
+
+    const stream = response.body;
+    if (!stream) {
+      throw new Error("The response stream is undefined");
+    }
+
+    if (response.status !== "200") {
+      throw new Error(`Failed to get chat completions: ${streamToString(stream)}`);
+    }
+
+    const sses = createSseStream(stream as IncomingMessage);
+    const functionArray: Array<any> = [];
+
+    for await (const event of sses) {
+
+      if (event.data === "[DONE]") {
+        continue;
+      }
+
+      for (const choice of (JSON.parse(event.data)).choices) {
```

Comment:
Should we give the output of parsing `event.data` a type to enable IDE intellisense?